File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = -1 if x < 0 else 1
        x = abs(x)
        c = 0
        while x != 0:
            d = x % 10
            c = c * 10 + d
            x //= 10
        c *= sign
        if c < -2**31 or c > 2**31 - 1:
            return 0

        return c

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int sign = 1;
        if(x > 0){
            sign = 1;
        }else{
            sign = -1;
        }
        long n = Math.abs(x);
        long digit = 0 , rev = 0;
        while(n > 0){
            digit = n % 10;
            rev = rev * 10 + digit;
            n /= 10;
        }
        rev *= sign;
       if (rev < Integer.MIN_VALUE || rev > Integer.MAX_VALUE) {
            return 0;
        }

        return (int) rev;
    }
}